org.eclipse.swtchart.export/src/org/eclipse/swtchart/export/images/ImageSupplier.java

```diff
@@ -15,52 +15,77 @@
 import org.eclipse.swt.graphics.GC;
 import org.eclipse.swt.graphics.Image;
 import org.eclipse.swt.graphics.ImageData;
+import org.eclipse.swt.graphics.ImageDataProvider;
 import org.eclipse.swt.graphics.ImageLoader;
-import org.eclipse.swt.graphics.Rectangle;
-import org.eclipse.swt.graphics.Transform;
+import org.eclipse.swt.graphics.PaletteData;
 import org.eclipse.swtchart.extensions.core.BaseChart;
 
 public class ImageSupplier {
 
+	/**
+	 * Save image data representation to a file with specific format (BMP, PNG or JPG)
+	 * @param imageData the image data representation
+	 * @param fileName the file (relative or full path)
+	 * @param format SWT.IMAGE_BMP, SWT.IMAGE_PNG or SWT.IMAGE_JPEG 
+	 */
 	public void saveImage(ImageData imageData, String fileName, int format) {
-
 		ImageLoader imageLoader = new ImageLoader();
 		imageLoader.data = new ImageData[]{imageData};
 		imageLoader.save(fileName, format);
 	}
 
+	/**
+	 * Return image data representation of provided chart
+	 * @param baseChart the chart from which is create image data representation
+	 * @return the image data representation
+	 */
 	public ImageData getImageData(BaseChart baseChart) {
-
-		Image image = new Image(baseChart.getDisplay(), baseChart.getBounds());
-		GC gc = new GC(image);
-		/*
-		 * Under macOS, the image is flipped.
-		 */
-		if(isMacOS()) {
-			Transform transform = new Transform(gc.getDevice());
-			transform.rotate(-180);
-			Rectangle bounds = image.getBounds();
-			transform.translate(-bounds.width, -bounds.height);
-			gc.setTransform(transform);
-			baseChart.print(gc);
-			transform.dispose();
-		} else {
-			baseChart.print(gc);
-		}
-		//
+		// Force to redraw chart immediately to be sure that any
+		// previous Shell dialog won't be a part of the copied image
+		baseChart.redraw();
+		baseChart.update();
+		// Chart size
+		int width = baseChart.getSize().x;
+		int height = baseChart.getSize().y;
+		// Get image data provider
+		ImageDataProvider imageDataProvider = getImageDataProvider(width, height);
+		// Copy chart into the image
+		Image image = new Image(baseChart.getDisplay(), imageDataProvider);
+		GC gc = new GC(baseChart);
+		gc.copyArea(image, 0, 0);
+		// Retrieve image data
+		ImageData imageData = image.getImageData(100);
+		// If generated image created from copy area has not same size as chart, scale image data
+		if(width != imageData.width || height != imageData.height) 
```

Comment:
I've actually introduced this in order to get an image the same size as the chart.
I don't know why but the chart image is twice the chart size after the call to copy area, it seems it's coming from SWT cocoa native code. But you're wright, it can be done in later steps.

Reply:
How do I apply these changes now ?
Do I apply these two points then commit and redo a PR ?